docs/okeps/okep-50088-evpn.md

```diff
@@ -0,0 +1,422 @@
+
+### Using EVPN as the Overlay for Tenant Networks
+
+With integrating into a customer's already exising TOR spine and leaf architecture, Geneve can be disabled, and network
+segmentation will still persist for east/west traffic due to VXLAN tunnels with EVPN. This is true for both IP-VRFs,
+and  MAC-VRFs. This reduces packet overhead for customers, while also providing some other advantages that come with
+EVPN, such as link redundancy and broadcast, unknown unicast, and multicast (BUM) traffic suppression.
+
+## Proposed Solution
+
+EVPN will continue to build upon the BGP support already implemented into OVN-Kubernetes using FRR. This support includes
+integration with an OVN-Kubernetes API as well as an FRR-K8S API for configuring BGP peering and routing for UDNs. FRR
+already supports EVPN and similar API resources will be leveraged to accomplish configuring FRR as the BGP/EVPN control
+plane. A new EVPN CRD will be introduced as the API necessary to enable EVPN for a cluster and selected UDNs.
+
+FRR's implementation relies on specific Linux constructs in order to use EVPN:
+
+1. A VRF device
+2. Linux Bridge enslaved to the VRF
+3. An SVI attached to the bridge (in most cases)
+4. A VXLAN device enslaved to the bridge
+5. A VTEP IP configured locally (generally on a loopback interface)
+
+These objects are required to be configured to interoperate with FRR EVPN, and therefore this enhancement proposes a new
+controller functionality in OVN-Kubernetes that will be used to manage objects 2-5. The VRF itself is already managed by
+OVN-Kubernetes as part of the UDN itself. Furthermore, due to the reliance of FRR on these Linux devices, there is no current
+way to natively integrate the data path of EVPN into OVN/OVS. This may change in the future, but for now all traffic
+needs to traverse the Linux Kernel Networking stack. Due to this constraint, this enhancement proposes EVPN support only
+for local gateway mode.
+
+### Workflow Description
+
+Tenants as well as admins are able to create UDNs for their namespace, however it requires admin access to be able to
+configure BGP and expose UDNs over BGP. This trend will continue with EVPN, where it will require admin access in order
+to enable EVPN for one or more UDNs. A typical workflow will be:
+
+1. Configure BGP peering via interacting with the FRR-K8S API for a given set of worker nodes.
+2. Create a primary Layer 2 or Layer 3 UDN.
+3. Configure EVPN CR to enable EVPN for this UDN.
+4. Create a RouteAdvertisements CR to specify what routes should be advertised via EVPN for this UDN.
+
+### API Details
+
+FRR-K8S may need to be extended to allow for configuring specific EVPN FRR configuration. A new cluster-scoped CRD for
+EVPN will be created within OVN-Kubernetes:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: EVPN
+metadata:
+  name: default
+spec:
+  networkSelector:
+    matchLabels:
+      k8s.ovn.org/metadata.name: blue
+  vtep:
+    cidr: 100.64.0.0/24
```

Comment:
If we have multiple EVPN yaml,    can this  `cidr: 100.64.0.0/24` be overlapped?

Reply:
Good point. Yes this cidr can overlap between CRs, it is a host specific subnet which is used to create VXLAN tunnels. I'll call this out in the enhancement.

Reply:
done